File: septiembre2015/sep15_java/src/sep15_java/Volumen.java
package sep15_java;

public abstract class Volumen {
    private String titulo;
    private String autor;
    private int ISBN;
    
    // Constructor
    public Volumen(String titulo, String autor, int ISBN) {
        this.titulo = titulo;
        this.autor = autor;
        this.ISBN = ISBN;
    }
    
    // Getter para ISBN
    public int getISBN() {
        return ISBN;
    }
    
    // Método abstracto que deben implementar las clases derivadas
    public abstract void ver();
    
    // Método protegido para que las clases hijas lo usen
    protected void verDatos() {
        System.out.print(titulo + " (" + autor + ") " + ISBN);
    }
}

File: septiembre2015/sep15_c/src/Volumen.cpp
#include "Volumen.h"

Volumen::Volumen(char* t, char* a, int isbn)
{
    titulo = new char[strlen(t) + 1];
    strcpy(titulo, t);
    autor = new char[strlen(a) + 1];
    strcpy(autor, a);
    is = isbn;
}

Volumen::~Volumen()
{
    delete[] titulo;
    delete[] autor;
}

Volumen::Volumen(const Volumen &v)
{
    titulo = new char[strlen(v.titulo) + 1];
    strcpy(titulo, v.titulo);
    autor = new char[strlen(v.autor) + 1];
    strcpy(autor, v.autor);
    is = v.is;
}

Volumen& Volumen::operator=(const Volumen &v)
{
    if (this != &v)
    {
        delete[] titulo;
        delete[] autor;

        titulo = new char[strlen(v.titulo) + 1];
        strcpy(titulo, v.titulo);
        autor = new char[strlen(v.autor) + 1];
        strcpy(autor, v.autor);
        is = v.is;
    }
    return *this;
}

void Volumen::verDatos() const
{
    cout << titulo << " (" << autor << ") " << is;
}
